usaspending_api/search/v2/views/spending_by_category_views/spending_by_category.py

```diff
@@ -150,7 +150,11 @@ def build_elasticsearch_search_with_aggregations(self, filter_query: ES_Q) -> Op
             group_by_agg_key_values = {"order": {"sum_field": "desc"}}
         else:
             # Get count of unique buckets; terminate early if there are no buckets matching criteria
-            bucket_count = get_number_of_unique_terms(filter_query, f"{self.category.agg_key}.hash")
+            # check if category is federal accounts, if so, do not use hash
+            if self.category.agg_key != "federal_accounts":
+                bucket_count = get_number_of_unique_terms(filter_query, f"{self.category.agg_key}.hash")
+            else:
+                bucket_count = get_number_of_unique_terms(filter_query, f"{self.category.agg_key}")
```

Comment:
I would change the `es_transaction_template` to have a hash field of type murmur3 instead of adding this bit of logic. The murmur3 plugin helps to speed up the cardinality check.

Reply:
Using the hash field on federal accounts will prevent the cardinality from returning accurately since the hash will flatten multiple federal accounts in an array into a single hash